cmd/hccore/hccore.go

```diff
@@ -0,0 +1,136 @@
+// command line interface to developing and testing holochain applications
+
+// Copyright (C) 2013-2017, The MetaCurrency Project (Eric Harris-Braun, Arthur Brock, et. al.)
+// Use of this source code is governed by GPLv3 found in the LICENSE file
+//---------------------------------------------------------------------------------------
+// command line interface to developing and testing holochain applications
+
+package main
+
+import (
+  "fmt"
+  "github.com/metacurrency/holochain/cmd"
+  "github.com/urfave/cli"
+  "os"
+
+  "bytes"
+)
+
+const (
+  defaultPort = "4141"
+)
+
+func getCurrentDirectoryOrExit() (string) {
+  dir, err := cmd.GetCurrentDirectory()
+  if err != nil {
+    cmd.Die("HC: hccore.go: getCurrentDirectory: could not find current directory. Weird. Exitting")
+  }
+
+  return dir
+}
+
+var debug bool
+var rootPath, devPath, name string
+var fromWhichSourceDirectory string
+  var noQuestions, compile bool
+
+func setupApp() (app *cli.App) {
+  app           = cli.NewApp()
+  app.Name      = "hccore"
+  app.Usage     = "tools for Holochain Core developers"
+  // app.Version = fmt.Sprintf("0.0.0 (holochain %s)", holo.VersionStr)
+  app.Flags     = []cli.Flag{
+
+  }
+
+  app.Commands  = []cli.Command{
+    {
+      Name:               "fromLocalFilesystem",
+      // Aliases:    []string{"branch"},
+      Usage:              "install a Holochain Core from a local directory (defaults to .)",
+      Flags:              []cli.Flag{
+          cli.StringFlag{
+            Name:          "sourceDirectory",
+            Usage:         "path to source files containing checked out git branch, defaults to current directory",
+            Value:         getCurrentDirectoryOrExit(),
+            Destination:   &fromWhichSourceDirectory,
+          },
+      },
+      Subcommands:    []cli.Command{
+        {
+          Name:           "install",
+          // Aliases:    []string{"branch"},
+          Usage:          "install the version of Holochain Core in '.' onto the host system",
+          Flags:          []cli.Flag{
+              cli.BoolFlag{
+                Name:          "noQuestions",
+                Usage:         "once the files are made available to Golang, should we compile them?",
+                Destination:   &noQuestions,
+              },
+              cli.BoolFlag{
+                Name:          "compile",
+                Usage:         "once the files are made available to Golang, should we compile them?",
+                Destination:   &compile,
+              },
+          },
+          Action:         func(c *cli.Context) error {
+              fmt.Printf  ("HC: core.fromLocalFilesystem.install: installing from        : %v\n",     fromWhichSourceDirectory)
+
+              err := os.Chdir(fromWhichSourceDirectory)
+              if err != nil {
+                fmt.Printf("HC: core.fromLocalFilesystem.install: could not change dir to: %v\n",     fromWhichSourceDirectory)
+                os.Exit(1)
```

Comment:
Yeah, this is from the filepath source: 


// Clean returns the shortest path name equivalent to path
// by purely lexical processing. It applies the following rules
// iteratively until no further processing can be done:
//
//	1. Replace multiple Separator elements with a single one.
//	2. Eliminate each . path name element (the current directory).
//	3. Eliminate each inner .. path name element (the parent directory)
//	   along with the non-.. element that precedes it.
//	4. Eliminate .. elements that begin a rooted path:
//	   that is, replace "/.." by "/" at the beginning of a path,
//	   assuming Separator is '/'.
//
// The returned path ends in a slash only if it represents a root directory,
// such as "/" on Unix or `C:\` on Windows.
//
// Finally, any occurrences of slash are replaced by Separator.
//
// If the result of this process is an empty string, Clean
// returns the string ".".
//
// See also Rob Pike, ``Lexical File Names in Plan 9 or
// Getting Dot-Dot Right,''
// https://9p.io/sys/doc/lexnames.html